adapters/socketio_win32.c

```diff
@@ -342,17 +355,48 @@ CONCRETE_IO_HANDLE socketio_create(void* io_create_parameters)
                     {
                         result->addrInfo->ai_addr = calloc(1, sizeof(struct sockaddr_in));
 
-                        result->port = socket_io_config->port;
-                        result->on_io_open_complete = NULL;
-                        result->dns_resolver = dns_resolver_create(result->hostname, result->port, NULL);
-                        result->on_bytes_received = NULL;
-                        result->on_io_error = NULL;
-                        result->on_bytes_received_context = NULL;
-                        result->on_io_error_context = NULL;
-                        result->io_state = IO_STATE_CLOSED;
-                        result->keep_alive = tcp_keepalive;
+                        if (result->addrInfo->ai_addr == NULL)
+                        {
+                            LogError("Failure allocating ai_addr");
+                            free(result->addrInfo);
+                            if (result->hostname != NULL)
+                            {
+                                free(result->hostname);
+                            }
+                            singlylinkedlist_destroy(result->pending_io_list);
+                            free(result);
+                            result = NULL;
+                        }
+                        else
+                        {
+                            result->dns_resolver = dns_resolver_create(result->hostname, result->port, NULL);
+
+                            if (result->dns_resolver == NULL)
+                            {
+                                LogError("Failure creating dns_resolver");
+                                free(result->addrInfo->ai_addr);
+                                free(result->addrInfo);
+                                if (result->hostname != NULL)
```

Comment:
technically the check whether it is `NULL` shouldn't be needed. it's a nop if it is `NULL`.

Reply:
We will need now, with the aux function I've added.